server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -122,30 +114,64 @@ public TemporalAccessor parse(String input) {
         }
 
         try {
-            return parser.parse(input);
+            return doParse(input);
         } catch (DateTimeParseException e) {
             throw new IllegalArgumentException("failed to parse date field [" + input + "] with format [" + format + "]", e);
         }
     }
 
+    private TemporalAccessor doParse(String input) {
+        if (parsers.size() > 1) {
+            for (DateTimeFormatter formatter : parsers) {
+                if (tryParseUnresolved(formatter, input) == true) {
+                    return formatter.parse(input);
+                }
+            }
+        }
+        return firstParser().parse(input);
```

Comment:
is this doing a second parse attempt with the first parser, if going through all the parsers did not work?

Reply:
yes, I guess we could throw an exception after the loop. WDYT?

Reply:
or just be sure via ctor check, that the size is never `0`? maybe just use an array instead of an array list, than the below check would always be a fixed size check as well?

Reply:
It should never be 0, if no parsers are provided, the printer will be used.
The only reason for this >1 check is to at least when there is only one parser (the regular, not composite pattern) it is not parsing the input twice.

Reply:
as discussed, will add an exception indicating that all parsers has failed. This won't mislead that the first parser failed (as it is at the moment)